impl/src/main/java/slack/SlackResourceImpl.java

```diff
@@ -32,44 +38,103 @@ public SlackResourceImpl(SlackConfig slackConfig) {
 
     @Override
     public Observable<String> getUserEmail(String userId) {
-        return fromCallable(() -> slack.methods().usersInfo(
-            UsersInfoRequest
-            .builder()
+        return callSlack(() -> {
+
+            final UsersInfoRequest slackRequest = UsersInfoRequest
+                .builder()
                 .user(userId)
                 .token(slackConfig.getApiToken())
-            .build())
-            .getUser()
-            .getProfile()
-            .getEmail());
+                .build();
+
+            return slack.methods().usersInfo(slackRequest);
+        }).flatMap(usersInfoResponse -> {
+            if (usersInfoResponse.isOk()) {
+                if (usersInfoResponse.getUser().getProfile().getEmail() == null) {
+                    LOG.warn("Could not fetch users email, maybe scope users:read:email is missing: ");
+                    return error(new IllegalStateException("Could not fetch user email from slack: " + usersInfoResponse.getError()));
+                }
+                return just(usersInfoResponse.getUser().getProfile().getDisplayName());
```

Comment:
We return display name but the method is named getEmail?

Reply:
Found that also and fixed. New commit soon